File: file-handling/handling-csv-files/visual-basic/csv_file_high_earners.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        SelectHighEarners()
    End Sub

    Sub SelectHighEarners()
        Dim fileName As String = "movies.csv"
        Dim highEarners As Integer = 0

        Using reader As System.IO.StreamReader = New System.IO.StreamReader(fileName)
            Dim record As String
            record = reader.ReadLine()

            While record IsNot Nothing
                Dim movieData As String() = record.Split(",")
                Dim revenue As Decimal = Decimal.Parse(movieData(5))
                If revenue > 200 Then
                    Dim title As String = movieData(0)
                    Console.WriteLine($"Movie name: {title}, Revenue: ${revenue} million")
                    highEarners = highEarners + 1
                End If
                record = reader.ReadLine()  'Read next record
            End While
        End Using

        Console.WriteLine($"There were {highEarners} high earning movies.")
    End Sub


End Module

File: file-handling/handling-csv-files/java/csv_file_high_earners.java
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science

Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

class HandlingCSVFiles
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args) {
        selectHighEarners();
    }


    // Select and display high earning movies over $200 million
    public static void selectHighEarners() {
        int highEarners = 0;

        // Uses try-with-resources to automatically close the readers after execution
        try (FileReader fr = new FileReader("movies.csv");
             BufferedReader reader = new BufferedReader(fr)) {

            String record = reader.readLine();

            while (record != null) {
                String[] movieData = record.split(",");
                double revenue =  Double.parseDouble(movieData[5]);

                // Check if the movie revenue was over $200 million
                if (revenue > 200) {
                    String title = movieData[0];
                    System.out.println("Movie name: " + title + ", Revenue: $" + revenue + " million");
                    highEarners = highEarners + 1;
                }

                record = reader.readLine(); // Read the next line
            }
        } // The readers are now closed
        catch (IOException e) {
            e.printStackTrace();
        }
        System.out.println("There were " + highEarners + " high earning movies.");
    }


}